app/services/appropriate_bodies/claim_an_ect/register_ect.rb

```diff
@@ -53,11 +53,16 @@ def update_itt_provider_name
       end
 
       def manage_teacher
-        @manage_teacher ||= ::Teachers::Manage.new(author:, teacher:, appropriate_body:)
+        @manage_teacher ||= ::Teachers::Manage.find_or_initialize_by(
+          trn: pending_induction_submission.trn,
+          trs_first_name: pending_induction_submission.trs_first_name,
+          trs_last_name: pending_induction_submission.trs_last_name,
+          event_metadata: { author:, appropriate_body: }
+        )
       end
 
       def teacher
-        @teacher ||= Teacher.find_or_initialize_by(trn: pending_induction_submission.trn)
+        @teacher ||= manage_teacher.teacher
```

Comment:
Was chatting with @avinhurry about memoization. I believe we can remove it here because `manage_teacher` keeps hold of the object, but generally, are we going for:
1. Do it only when required 
2. Everything, everywhere, all at once

Reply:
Thanks @peterdavidhamilton, added a commit that fixes this.